hastory-cli/src/Hastory/Cli/Internal.hs

```diff
@@ -51,3 +52,17 @@ runDb dbAction = do
   SQL.runSqlite (T.pack . toFilePath $ hDb) $ do
     SQL.runMigration migrateAll
     dbAction
+
+runDb' ::
+     (MonadReader Settings m, MonadUnliftIO m)
+  => ReaderT SqlBackend (NoLoggingT (ResourceT m)) a
+  -> m a
+runDb' dbAction = do
+  sets <- ask
+  case runReaderT histDb sets of
+    Nothing -> liftIO $ die "Unable to locate histDb"
```

Comment:
Here is my thought process on this:

I want to define a function `runDb'` that is as close as possible to `runDb`, but that does not have the `MonadThrow` constraint. The `MonadThrow` constraint is used by `histDb` solely because it is used by [parseRelFile](https://www.stackage.org/haddock/lts-16.16/path-0.7.0/Path-Posix.html#v:parseRelFile). By using `runReaderT histDb sets`, we are specializing `histDb` to `ReaderT Settings Maybe (Path Abs File)`.

All of this to say that if `parseRelFile` fails, the connection info is wrong, which means we can't connect to the database, which means we have to fail since there is no recovery from that. 

Now that I think about it, I can just let `parseRelFile` fail in IO, and then it will take care of the exiting itself.

```hastory
runDb' ::
     (MonadReader Settings m, MonadUnliftIO m)
  => ReaderT SqlBackend (NoLoggingT (ResourceT m)) a
  -> m a
runDb' dbAction = do
  sets <- ask
  hDb <- liftIO $ runReaderT histDb sets
  ensureDir $ parent hDb
  SQL.runSqlite (T.pack . toFilePath $ hDb) $ do
    SQL.runMigration migrateAll
    dbAction
```

This might be something that can be revisited in #34